packages/shared/src/telemetry/events/component-mounted.ts

```diff
@@ -0,0 +1,22 @@
+const EVENT_COMPONENT_MOUNTED = 'COMPONENT_MOUNTED' as const;
+
+type EventComponentMounted = {
```

Comment:
Thanks for the feedback! I initially refrained from creating a type like you've outlined to keep the implementation simple. When using the helper, implementors would get type errors from the call to `telemetry.record()` if the shape isn't correct.

The extra layer of helper types here creates some complexity and cognitive overhead that I was trying to avoid, but let me take another look!

Reply:
Pushed an update. I've introduced a single type to be used as the return value from the factory methods. I think this strikes a good balance between too much typing boilerplate while still ensuring correctness when we are adding new events. Let me know what you think 👍